src/types/smartSearch.ts

```diff
@@ -161,6 +175,45 @@ export interface SubQueryFilterConfig {
     query_id: number;
 }
 
+interface TaskTimeFrameBefore {
+    before: string;
+}
+
+interface TaskTimeFrameAfter {
+    after: string;
+}
+
+interface TaskTimeFrameBetween {
+    after: string;
+    before: string;
+}
+
+type TaskTimeFrame = boolean
+        | TaskTimeFrameAfter
+        | TaskTimeFrameBefore
+        | TaskTimeFrameBetween;
```

Comment:
Nice! This actually (much better) expresses in code what I was trying to say the other day, which is that even without the `boolean` this is still a union type.

Reply:
Yeah I figured this would help to find any locations where this was not the case, in order to eliminate any `{}`. Obviously didn't work out as well as I had hoped though.